imagej/imagej-ops2-tutorial/docs/OpCurrency.md

```diff
@@ -0,0 +1,130 @@
+# Writing Small, Easy, Modular Ops
+
+SciJava Ops is designed for modularity, extensibility, and granularity - by writing your Ops on the smallest functional elements possible, you'll allow SciJava Ops to lift your Op to new image types, enable parallelism, and more without any additional effort!
+
+## Element-wise Ops
+
+Simple pixel-wise operations like addition, inversion, and more can be written on a single pixel (i.e. `RealType`) - therefore, ImageJ Ops2 takes care to automagically adapt pixel-wise Ops across a wide variety of image types. If you would like to write a pixel-wise Op, we recommend the following structure.
+
+```java
+/**
+ * A simple pixelwise Op
+ * 
+ * @implNote op names="my.op"
+ */
+class <T extends RealType<T>> MyPixelwiseOp implements Computers.Arity2<T, T, T> {
+	@Override
+        public void compute(final T in1, final T in2, final T out) {
+            ... pixelwise computation here ...
+        }
+}
+```
+
+If you then have ImageJ Ops2, the following Op call will match your pixel-wise Op using ImageJ Ops2's lifting mechanisms:
+
+```java
+ArrayImg<UnsignedByteType> in1 = ...
+ArrayImg<UnsignedByteType> in2 = ...
+ArrayImg<UnsignedByteType> out = ...
+
+ops.op("my.op").arity2().input(in1, in2).output(out).compute();
+```
+
+A similar vein of thought works for simple `List`s and `Array`s - if you have an Op that produces a `Double` from another `Double`, there's no need to write a wrapper to work on `Double[]`s - Ops will do that for you!
+
+```java
+/**
+ * A simple pixelwise Op
+ *
+ * @implNote op names="my.op"
+ */
+class MyPixelwiseOp implements Function<Double, Double> {
```

Comment:
PixelwiseOp is used above... `MyCollectionOp`?

Reply:
Renamed in f4f121e2 to `MyElementOp` - I didn't want any confusion with `OpCollection`s.